prov/rxd/src/rxd_proto.h

```diff
@@ -166,8 +167,7 @@ struct rxd_data_hdr {
 };
 
 struct rxd_rma_hdr {
-	uint64_t		iov_count;
-	struct ofi_rma_iov	rma[];	
+	struct ofi_rma_iov	rma[RXD_IOV_LIMIT];
```

Comment:
Weirdly, yes, I know what you're saying.
The size of the header is coming from the iov_count which is either presumed to be 1 without the sar_hdr or equal to sar_hdr->iov_count. You're right, we could calculate it at the rx side but (a) we have that extra space anyway, and (b) that makes unpacking the headers rather tricky for a couple of reasons. So I'm inclined to ignore your ramblings this time.
And with regards to the rma[] note. I had it like that, but my compiler complains about a variable sized field in an otherwise empty struct. blah blah. That was my way around it, but if you know a better way, I'm all ears.